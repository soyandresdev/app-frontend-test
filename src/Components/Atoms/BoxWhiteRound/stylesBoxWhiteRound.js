import styled from "styled-components";

export const BoxWhiteRound = styled.section`
  background: #fff;
  border-radius: 13px;
  padding: 20px;
  height: 100%;
  border: 1px solid #eee;
`;
