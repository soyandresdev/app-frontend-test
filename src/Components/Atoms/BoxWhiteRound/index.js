import React from "react";
import PropTypes from "prop-types";

import { BoxWhiteRound as BoxWhiteRoundStyles } from "./stylesBoxWhiteRound.js";

export default function BoxWhiteRound({ children }) {
  return <BoxWhiteRoundStyles>{children}</BoxWhiteRoundStyles>;
}

BoxWhiteRound.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]).isRequired
};
