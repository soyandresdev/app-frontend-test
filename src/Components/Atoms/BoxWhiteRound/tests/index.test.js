import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import BoxWhiteRound from "../index";
import { BoxWhiteRound as BoxWhiteRoundStyles } from "../stylesBoxWhiteRound";

describe("Test component [BoxWhiteRound]", () => {
  const Props = {
    children: "<h1>hola</h1>"
  };
  const wrapper = shallow(<BoxWhiteRound {...Props} />);

  it("Renders correctly defaultProps [BoxWhiteRound]", () => {
    const rendered = renderer.create(<BoxWhiteRound {...Props} />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it("Renders correctly [BoxWhiteRound]", () => {
    const rendered = renderer.create(<BoxWhiteRound {...Props} />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it("Test init props required [BoxWhiteRound]", () => {
    expect(wrapper.find(BoxWhiteRoundStyles).length).toBe(1);
    expect(
      wrapper
        .find(BoxWhiteRoundStyles)
        .at(0)
        .text()
    ).toBe(Props.children);
  });
});
