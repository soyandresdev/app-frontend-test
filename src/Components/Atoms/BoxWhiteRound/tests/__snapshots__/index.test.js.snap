// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component [BoxWhiteRound] Renders correctly [BoxWhiteRound] 1`] = `
<section
  className="sc-bdVaJa bRZyQZ"
>
  &lt;h1&gt;hola&lt;/h1&gt;
</section>
`;

exports[`Test component [BoxWhiteRound] Renders correctly defaultProps [BoxWhiteRound] 1`] = `
<section
  className="sc-bdVaJa bRZyQZ"
>
  &lt;h1&gt;hola&lt;/h1&gt;
</section>
`;
